homeassistant/components/caldav/todo.py

```diff
@@ -156,7 +157,15 @@ async def async_update_todo_item(self, item: TodoItem) -> None:
         vtodo = todo.icalendar_component  # type: ignore[attr-defined]
         vtodo["SUMMARY"] = item.summary or ""
         if status := item.status:
-            vtodo["STATUS"] = TODO_STATUS_MAP_INV.get(status, "NEEDS-ACTION")
+            if status == TodoItemStatus.COMPLETED:
+                try:
+                    await self.hass.async_add_executor_job(
+                        partial(todo.complete, completion_timestamp=datetime.now()),
+                    )
+                except (requests.ConnectionError, DAVError) as err:
+                    raise HomeAssistantError(f"CalDAV complete error: {err}") from err
```

Comment:
Regarding the test failure, the test `test_update_item_failure`  has a [match](https://github.com/home-assistant/core/blob/ad0ee7d781c78755b743324056004180e6796b2e/tests/components/caldav/test_todo.py#L547) based on the specific string in this exception that needs to be updated.

Reply:
I'll need help regarding tests.
They are failing because the "COMPLETED" date is not expected, but as that date is the current one, I cannot hardcode it.

How can that test be adapted?

Thanks.

Reply:
One approach I would recommend is to use the `freezegun` library which will patch the system date, allowing you to hard code a string in the test. There is an example in the calendar tests like https://github.com/home-assistant/core/blob/2c91b312331ce76a078da2c30c60ca0fa7cf82f6/tests/components/caldav/test_calendar.py#L437